exporter/loadbalancingexporter/exporter_test.go

```diff
@@ -521,6 +524,110 @@ func TestNoTracesInBatch(t *testing.T) {
 	}
 }
 
+func TestRollingUpdatesWhenConsumeTraces(t *testing.T) {
+	// this test is based on the discussion in the following issue for this exporter:
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1690
+	// prepare
+
+	// simulate rolling updates, the dns resolver should resolve in the following order
+	// ["127.0.0.1"] -> ["127.0.0.1", "127.0.0.2"] -> ["127.0.0.1"]
+	res, err := newDNSResolver(zap.NewNop(), "service-1", "")
+	require.NoError(t, err)
+
+	counter := 0
+	resolve := [][]net.IPAddr{
+		{
+			{IP: net.IPv4(127, 0, 0, 1)},
+		}, {
+			{IP: net.IPv4(127, 0, 0, 1)},
+			{IP: net.IPv4(127, 0, 0, 2)},
+		}, {
+			{IP: net.IPv4(127, 0, 0, 2)},
+		},
+	}
+	res.resolver = &mockDNSResolver{
+		onLookupIPAddr: func(context.Context, string) ([]net.IPAddr, error) {
+			defer func() {
+				counter++
+			}()
+
+			if counter > 2 {
+				return resolve[2], nil
+			}
+
+			return resolve[counter], nil
+		},
+	}
+	res.resInterval = 10 * time.Millisecond
+
+	config := &Config{
+		Resolver: ResolverSettings{
+			DNS: &DNSResolver{Hostname: "service-1", Port: ""},
+		},
+	}
+	params := component.ExporterCreateParams{Logger: zap.NewNop()}
+	p, err := newExporter(params, config)
+	require.NotNil(t, p)
+	require.NoError(t, err)
+
+	p.res = res
+
+	var counter1, counter2 int
+	defaultExporters := map[string]component.TracesExporter{
+		"127.0.0.1": &mockTracesExporter{
+			ConsumeTracesFn: func(ctx context.Context, td pdata.Traces) error {
+				counter1 += 1
+				// simulate an unreachable backend
+				time.Sleep(10 * time.Second)
+				return nil
+			},
+		},
+		"127.0.0.2": &mockTracesExporter{
+			ConsumeTracesFn: func(ctx context.Context, td pdata.Traces) error {
+				counter2 += 1
+				return nil
+			},
+		},
+	}
+
+	// test
+	err = p.Start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+	defer p.Shutdown(context.Background())
+	// ensure using default exporters
+	p.exporters = defaultExporters
+	p.res.onChange(func(endpoints []string) {
+		p.exporters = defaultExporters
+	})
+
+	// keep consuming traces every 2ms
+	go func() {
+		ticker := time.NewTicker(2 * time.Millisecond)
+		for {
+			select {
+			case <- ticker.C:
+				go p.ConsumeTraces(context.Background(), randomTraces())
```

Comment:
Do you need this to be another go routine? With this change, this shouldn't block long anymore, right?

Reply:
yes, you're right. I should remove the `go`.

Reply:
ah. after a deep thought, I found that it does need the go. because I use time.Sleep in ConsumeTraces to simulate unreachable backends, if I don't use go routine, this loop will stuck there and stop consuming traces.